### ISSUE TITLE
Lint (pylint, flake8) fail when file accessed is under directory path containing spaces

### ISSUE THREAD

**Issue by mtjoseph** [DonJayamanne/pythonVSCode#793]
## Environment data
VS Code version:  1.10.1 (1.10.1)
Python Extension version:  0.5.9
Python Version:   3.6.0
OS and version:    Mac OS Sierra 10.12.4 Beta (16E175b)

## Actual behavior
Enabled pylint or flake8, upon saving, or opening (or linting activation), error thrown in Output window:
Linting with flake8 failed.
Command failed: flake8 --format=%(row)d,%(col)d,%(code)s,%(code)s:%(text)s /private/tmp/project/dir with spaces/a/b/c/anotherfile.py

** Note the un-escaped space in directory path.

## Expected  behavior
Handle directory paths with spaces, escaping the path as needed.

## Steps to reproduce:  
- Outside VSCode
-  mk "/tmp/project/dir with spaces/a/b/c"
- touch "/tmp/project/dir with spaces/a/b/c/anotherfile.py"

With linting enabled:
- Open project folder in VSCode
- Open anotherfile.py

error message displayed: 
Linting with flake8 failed.
Command failed: flake8 --format=%(row)d,%(col)d,%(code)s,%(code)s:%(text)s /private/tmp/project/dir with spaces/a/b/c/anotherfile.py

** Note: The error is always thrown when the file is opened. It causes erratic editing behavior afterwards, sometimes, editing can be continued, while others I have to restart VSCode as the editor becomes unresponsive.

## Settings  
Your settings.json:   
    // Whether to lint Python files.
    "python.linting.enabled": true,

    // Whether to lint Python files using pylint.
    "python.linting.pylintEnabled": false,

    // Whether to lint Python files using flake8
    "python.linting.flake8Enabled": true,

    // Whether to lint Python files using pydocstyle
    "python.linting.pydocstyleEnabled": false,

    // Whether to lint Python files using mypy.
    "python.linting.mypyEnabled": false,

    // Whether to lint Python files when modified.
    "python.linting.lintOnTextChange": false,

    // Whether to lint Python files when saved.
    "python.linting.lintOnSave": true,

    // Controls the maximum number of problems produced by the server.
    "python.linting.maxNumberOfProblems": 50,

    // Arguments passed in. Each argument is a separate item in the array.
    "python.linting.pylintArgs": [],

    // Arguments passed in. Each argument is a separate item in the array.
    "python.linting.flake8Args": [],

## Logs
```
Linting with flake8 failed.
Command failed: flake8 --format=%(row)d,%(col)d,%(code)s,%(code)s:%(text)s /private/tmp/project/dir with spaces/a/b/c/anotherfile.py

##########Linting Output - flake8##########
4,1,F401,F401:'re' imported but unused
5,1,F401,F401:'sys' imported but unused
9,1,E265,E265:block comment should start with '# '
10,1,E302,E302:expected 2 blank lines, found 1 
```

Output from ```Console window``` (Help->Developer Tools menu)
```

There was an error in running the linter 'flake8'                    
e.doShow @ messageService.ts:126
e.show @ messageService.ts:105
(anonymous function) @ mainThreadMessageService.ts:64
n.Class.derive._oncancel @ winjs.base.raw.js:164
1t.showMessage @ mainThreadMessageService.ts:34
t.$showMessage @ mainThreadMessageService.ts:28
e.handle @ abstractThreadService.ts:34
s @ ipcRemoteCom.ts:275f @ ipcRemoteCom.ts:231
_combinedTickCallback @ internal/process/next_tick.js:67
_tickCallback @ internal/process/next_tick.js:98
```

**Comment by DonJayamanne** [comment]
I believe I have fixed this issue, will release a new version soon and let you know.

**Comment by DonJayamanne** [comment]
Moving this issue, has it is similar to https://github.com/DonJayamanne/pythonVSCode/issues/838 (and that's where I've started fixing this issue)

**Comment by janhancic** [comment]
Coming here from #838 ... none of my project or files in it contain spaces so not sure if this is the same issue?

**Comment by janhancic** [comment]
Ah sorry, I was confused by github for a second and thought you closed #838 as a duplicate of this. Nevermind me :)

**Comment by mtjoseph** [comment]
No problem. I will update and retest and mark the bug accordingly.

Thanks,
Maurice

**Comment by mtjoseph** [comment]
Problem fixed in v0.6.2
